bottlerocket/agents/src/bin/vsphere-k8s-cluster-resource-agent/vsphere_k8s_cluster_provider.rs

```diff
@@ -505,6 +535,15 @@ async fn write_vsphere_clusterspec(
     let vcenter_workload_folder = config.vcenter_workload_folder.to_owned();
     let vcenter_resource_pool = config.vcenter_resource_pool.to_owned();
 
+    // Retrive associated cluster license token value from secret manager.
+    // The license token name is <cluster_name>-token
+    let license_token_name = format!("{}-token", &cluster_name);
+    let eksa_entended_license_parsed_vlaues =
+        get_secret_values(secret_manager_client, VMWARE_CLUSTER_LICENSE_TOKEN).await?;
+    let eksa_entended_license_vlaue = eksa_entended_license_parsed_vlaues[&license_token_name]
```

Comment:
There are a lot of typos in these names. Also, back to my comment about abstracting this a bit from secrets manager. I'd love for this function to just be calling a `get_license_token(cluster_name)` to make this less specific in this path here.

Reply:
Thanks for the comment! When I implemented `get_secret_values ` function, I think this function would be useful for other agent images if they need to get secret values from secret manager. I agree that  `get_license_token(cluster_name)` would be more clear, so i think I can do
```
fn get_license_token(cluster_name) {
     let license_token_name = format!("{}-token", &cluster_name);
     let eksa_entended_license_parsed_vlaues =
        get_secret_values(secret_manager_client, VMWARE_CLUSTER_LICENSE_TOKEN).await?;
    let eksa_entended_license_vlaue = eksa_entended_license_parsed_vlaues[&license_token_name]
}
```